src/menu/ide_menu.rs

```diff
@@ -1487,4 +1493,50 @@ mod tests {
         menu.update_values(&mut editor, &mut completer);
         assert!(menu.menu_string(2, true).contains("おは"));
     }
+
+    #[test]
+    fn test_menu_create_value_string_starting_with_multibyte_char() {
+        // https://github.com/nushell/nushell/issues/15938
+        let mut completer = FakeCompleter::new(&["验abcdef/", "abcdef/"]);
+        let mut menu = IdeMenu::default().with_name("testmenu");
+        menu.working_details = IdeMenuDetails {
+            cursor_col: 50,
+            menu_width: 50,
+            completion_width: 50,
+            description_width: 50,
+            description_is_right: true,
+            space_left: 50,
+            space_right: 50,
+            description_offset: 50,
+            shortest_base_string: String::new(),
+        };
+        let mut editor = Editor::default();
+
+        editor.set_buffer("ac".to_string(), UndoBehavior::CreateUndoPoint);
+        menu.update_values(&mut editor, &mut completer);
+        assert!(menu.menu_string(10, true).contains("验"));
+    }
+
+    #[test]
+    fn test_menu_create_value_string_long_unicode_string() {
+        // https://github.com/nushell/reedline/pull/918
+        let mut completer = FakeCompleter::new(&["验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验验abcdef/", "abcdef/"]);
+        let mut menu = IdeMenu::default().with_name("testmenu");
+        menu.working_details = IdeMenuDetails {
+            cursor_col: 50,
+            menu_width: 50,
+            completion_width: 50,
+            description_width: 50,
+            description_is_right: true,
+            space_left: 50,
+            space_right: 50,
+            description_offset: 50,
+            shortest_base_string: String::new(),
+        };
```

Comment:
I think you just need to assign `completion_width`. If you make it something small (e.g., 5), you won't have to make the suggestion above super long
```suggestion
        menu.working_details.completion_width = 5;
```

Reply:
That's true changing `completion_width` is another way to trigger the crash.

I wanted to clean it up a different way though - I didn't want to make the string shorter as I feel like sending a long string through the code is like the core of the test. It should be a stress test regardless of defaults. With that in mind I changed it to this -

`FakeCompleter::new(&[&("验".repeat(205) + "abc/")])`

And then I got rid of setting the `completion_width` at all, as 205 long will be way longer than any reasonable default even if someone tweaks things in the future.

This also keeps the test code the same between ide and columnar, since in columnar there's no `completion_width` to set. You mentioned the duplication will be removed in future - doing that will be easy and obvious if the duplicated test code is identical.

Also if anyone adds truncation to columnar, or changes the defaults in ide, this will continue to stress test it.

Reply:
Actually I needed to set `completion_width` or the test never passes - I guess there's no default value. I still set it to 50 though for the above reasons.

Reply:
Gotcha. `.repeat` is fine too